src/Form/UpdateMediaItemForAccount.php

```diff
@@ -0,0 +1,170 @@
+    $this->mediaStorage = $entityTypeManager->getStorage('media');
+    $this->accountsStorage = $entityTypeManager->getStorage('media_mpx_account');
+    $this->updateVideoItemService = $updateVideoItem;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('media_mpx.service.update_video_item')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $accounts = $this->accountsStorage->loadMultiple();
+    $account_opts = [];
+
+    foreach ($accounts as $account) {
+      $account_opts[$account->id()] = $account->label();
+    }
+
+    $form['account'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Accounts'),
+      '#description' => $this->t('Choose the account that owns the video you want to update.'),
+      '#options' => $account_opts,
+      '#required' => TRUE,
+    ];
+    $form['guid'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('guid'),
+      '#placeholder' => 'Type the GUID of the mpx item you want to import.',
+      '#required' => TRUE,
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Update item'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Submit handler for the 'media_mpx_asset_sync_single_by_account_guid' form.
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $account = $form_state->getValue('account');
+    $guid = $form_state->getValue('guid');
+
+    if (!$media_items = $this->loadMediaEntitiesWithMatchingGuid($guid)) {
+      $this->messenger()->addError($this->t('There are no video items with the entered GUID.'));
+      return;
+    }
+
+    // Of all the possible videos, get the one linked to the selected account,
+    // since guids can vary between different accounts.
+    if (!$account_linked_video = $this->getMediaItemLinkedToAccount($media_items, $account)) {
+      $this->messenger()->addError($this->t('The guid entered is not associated to the selected account.'));
+      return;
+    }
+
+    $updateRequest = UpdateVideoItemRequest::createFromMediaEntity($account_linked_video);
+    $this->updateVideoItemService->execute($updateRequest);
+    // @todo: add exception and response handling for the update service.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'media_mpx_asset_sync_single_by_account_guid';
+  }
+
+  /**
+   * Loads all the media entities with a given guid.
+   *
+   * @param string $guid
+   *   The guid to filter by.
+   *
+   * @return \Drupal\media\Entity\Media[]
+   *   The mpx Media entities matching the given guid
+   */
+  private function loadMediaEntitiesWithMatchingGuid(string $guid): array {
+    $media_ids = $this->mediaStorage->getQuery()
+      ->condition('field_mpx_guid', $guid)
```

Comment:
This field is defined by sites, so the name may be different. I think this would need to be `getMetadata(...)` from the media source plugin.

Reply:
It's taken a while, as I thought this was configured, and went around in circles wondering why fieldMap was coming with empty results all the time. oh well. 🤦‍♂  Done now :).